Source/Graphics/Backend/CommandList.cpp

```diff
@@ -0,0 +1,91 @@
+#pragma once
+
+#include <Zmey/Graphics/Backend/CommandList.h>
+#include <Zmey/Graphics/Managers/MaterialManager.h>
+#include <Zmey/Graphics/Backend/GraphicsPipelineState.h>
+#include <Zmey/Graphics/Backend/BackendResourceSet.h>
+#include <Zmey/Modules.h>
+
+namespace Zmey
+{
+namespace Graphics
+{
+namespace Backend
+{
+namespace
+{
+bool IsResourceSetInPipeline(GraphicsPipelineState* pipeline, ResourceSetType set)
+{
+	return std::find(
+		pipeline->Desc.ResourceTable.ResourceSets.begin(),

```

Comment:
If we use only ResourceStets why we take pipelinestate ?

Reply:
As it is longer to write IsResourceSetInPipeline(pipeline->Desc.ResourceTable, ..)
instead of just pipeline. And this will be written more times. This function was written once only. Also in future there could be something else we need from the pipeline state